proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -401,14 +409,18 @@ as the result of changing some of [`internal`]'s components.
 
 * Let `components` be `"alpha"` and the names of the channels in `space`.
 
-* If any key in `keys` is not the name of a channel in `components`, throw an
-  error.
+* If any key in `keys` is not the name of a channel in `components`:
+
+  * If `initialSpace` is a [legacy color space] and `spaceSetExplicitly` is
+    false, emit a deprecation warning named `color-4-api`.
+
+  * Otherwise, throw an error.
```

Comment:
I think we can comfortably start throwing an error without updating the spec here and call it a potentially-breaking bug fix. We have a general rule that if users violate the TypeScript types, the behavior is undefined. We'll try to deprecate in cases where it's very likely that the types were being violated in good faith (like passing `null` instead of `undefined` to optional parameters), but I think it's pretty unlikely that anyone is actually running `new SassColor({hue: ..., red: ...})` and expecting it to work.

Reply:
Addressed in https://github.com/sass/sass/pull/3704/commits/7308783af5646e553c35b4944216bcc78b781a44

Reply:
@nex3 This brings a related point up, then. Currently the proposed types for `change` do allow for `myColor.change({hue: ..., red: ..., x: ...})` through the overload without a space defined. We could limit the types by removing that overload, and then setting the `space` to optional in all the other types. I think this also communicates the interface a bit more directly as well.